creusot/src/backend/logic.rs

```diff
@@ -185,9 +185,9 @@ pub(crate) fn lower_logical_defn<'tcx, N: Namer<'tcx>>(
     );
 
     if has_axioms {
-        if sig.uses_simple_triggers() {
+        if sig.uses_simple_triggers() && !sig_contract.contract.variant.is_empty() {
```

Comment:
why is this second condition needed? we don't seem to require it in the non-simpl-trigger case?

Reply:
The current implementation only uses `lim` functions when there is a `variant` since otherwise it uses and interpreted why3 `function`